pcb/check_kicad_mod.py

```diff
@@ -65,32 +76,47 @@
         printer.green('rotated footprint by {deg} degrees'.format(deg=int(args.rotate)))
         
     n_violations = 0
+    
+    no_warnings = True
+    
+    output = []
+    
+    first = True
+    
     for rule in all_rules:
+    
         rule = rule(module,args)
-        if rule.check():
-            #this is the first violation
-            if n_violations == 0:
-                printer.green('checking module: %s' % module.name)
+        
+        error = rule.check()
+        
+        if error:
             n_violations += 1
+        
+        # Any messages (either warnings OR errors)
+        if error or len(rule.verbose_message) > 0:
+        
+            no_warnings = False
+        
+            if first:
+                first = False
+                printer.green('checking module: %s' % module.name)
+                        
             printer.yellow('Violating ' + rule.name, indentation=2)
             if args.verbose:
-                printer.light_blue(rule.description, indentation=4, max_width=100)
                 if len(rule.verbose_message)>0:
-                    vm=rule.verbose_message.split('\n');
-                    for v in vm:
+                    printer.light_blue(rule.description, indentation=4, max_width=100)
+                for msg in rule.verbose_message:
+                    for v in msg.split("\n"):
                         printer.blue(v, indentation=6, max_width=100)
-
-                # example of customized printing feedback by checking the rule name
-                # and a specific variable of the rule
-                #if rule.name == 'Rule 6.6' and len(rule.f_courtyard_all) == 0:
-                #    printer.red('No courtyard line found in the module', indentation=4)
+                
         if args.fix:
             rule.fix()
             if args.verbose:
                 if len(rule.fix_verbose_message)>0:
                     vm=rule.fix_verbose_message.split('\n');
```

Comment:
should be `vm=rule.fix_message;` as `fix_message` is now a list and not a string!